cmd/juju/cloud/addcredential.go

```diff
@@ -523,9 +581,18 @@ func enterFile(name, descr string, p *interact.Pollster, expanded, optional bool
 	return string(contents), errors.Trace(err)
 }
 
-func shouldFinalizeCredential(provider environs.EnvironProvider, cred jujucloud.Credential) bool {
+func ShouldFinalizeCredential(provider environs.EnvironProvider, cred jujucloud.Credential) bool {
 	if finalizer, ok := provider.(environs.RequestFinalizeCredential); ok {
 		return finalizer.ShouldFinalizeCredential(cred)
 	}
 	return false
 }
+
+func ValidCloudRegion(aCloud *jujucloud.Cloud, region string) error {
```

Comment:
I'm a fan of adding explicit tests for exported functions.

Reply:
This does not need to be exported.